### ISSUE TITLE
For circle loss, what if similarity_matrix[positive_matrix] is empty?

### ISSUE THREAD

**Issue by liamsun2019** [layumi/Person_reID_baseline_pytorch#333]
Hi author，
As for convert_label_to_similarity in circle loss, , there's every chance that positive_matrix be all False. Therefore, similarity_matrix[positive_matrix] will have no element at all, which incurs error in following circle loss computation. How can we handle such case? Any suggestions are appreciated. Thanks.

For instance:
feat = nn.functional.normalize(torch.rand(256, 64, requires_grad=True))
#0~255
lbl = torch.linspace(0, 255, steps=256).int()
inp_sp, inp_sn = convert_label_to_similarity(feat, lbl)

Above lbl will generate an empty similarity_matrix[positive_matrix] in convert_label_to_similarity.

**Comment by layumi** [comment]
Hi @liamsun2019 

I am not sure whether I understand your question. 

It seems to be Okay. 

Since you use `lbl = torch.linspace(0, 255, steps=256).int()`, there are not same label in a batch . 

The similarity metric is a identity matrix.
